pipelines/internal/commands/query/query.go

```diff
@@ -37,13 +38,28 @@ func Invoke(ctx context.Context, service *genomics.Service, project string, argu
 	if *filter != "" {
 		call = call.Filter(*filter)
 	}
-	resp, err := call.Do()
-	if err != nil {
-		return err
+
+	if *limit == 0 {
+		return nil
 	}
 
-	for _, operation := range resp.Operations {
-		fmt.Println(operation.Name)
+	pageToken := ""
+	listed := uint(0)
+	for {
+		resp, err := call.PageToken(pageToken).Do()
+		if err != nil {
+			return err
+		}
+
+		for _, operation := range resp.Operations {
+			fmt.Println(operation.Name)
+			if listed++; listed == *limit {
+				return nil
+			}
+		}
+
+		if pageToken = resp.NextPageToken; pageToken == "" {
```

Comment:
It's odd to write side effects inside the conditionals like this.

if resp.NextPageToken == "" {
  return nil
}
pageToken = resp.NextPageToken

is more idiomatic and clear